test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -19,13 +87,273 @@
+    initInfo.creds.pLocalPassword = ( uint8_t * ) "localPassword";
+    initInfo.creds.pRemoteUsername = ( uint8_t * )  "remoteUsername";
+    initInfo.creds.pRemotePassword = ( uint8_t * ) "remotePassword";
+    initInfo.creds.pCombinedUsername = ( uint8_t * ) "combinedUsername";
+    initInfo.isControlling = 1;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_OK,
+                       result );
+    TEST_ASSERT_EQUAL( localCandidateArray,
+                       context.pLocalCandidates );
+    TEST_ASSERT_EQUAL( remoteCandidateArray,
+                       context.pRemoteCandidates );
+    TEST_ASSERT_EQUAL( candidatePairArray,
+                       context.pCandidatePairs );
+    TEST_ASSERT_EQUAL( transactionIdStore,
+                       context.pStunBindingRequestTransactionIdStore );
+    TEST_ASSERT_EQUAL( testRandomFxn,
+                       context.cryptoFunctions.randomFxn );
+    TEST_ASSERT_EQUAL( testCrc32Fxn,
+                       context.cryptoFunctions.crc32Fxn );
+    TEST_ASSERT_EQUAL( testHmacFxn,
+                       context.cryptoFunctions.hmacFxn );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate fail functionality for Bad Parameters.
+ */
+void test_iceAddHostCandidate_BadParams( void )
+{
+    IceContext_t context;
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    result = Ice_AddHostCandidate( NULL,
+                                   &( endPoint ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   NULL );
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate fail functionality for Max Candidate Threshold.
+ */
+void test_iceAddHostCandidate_MaxCandidateThreshold( void )
+{
+    IceContext_t context;
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    context.numLocalCandidates = 1000;
+    context.maxLocalCandidates = 1000;
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   &( endPoint ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_MAX_CANDIDATE_THRESHOLD,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate functionality.
+ */
+void test_iceAddHostCandidate( void )
+{
+    IceContext_t context = { 0 };
+    IceCandidate_t localCandidates[5];
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    context.pLocalCandidates = &( localCandidates[ 0 ] );
+    context.maxLocalCandidates = 5;
+    context.numLocalCandidates = 0;
+
+    endPoint.isPointToPoint = 1;
+    const uint8_t * ipAddress = ( uint8_t * ) "192.168.1.100";
```

Comment:
Put all the declarations in the beginning, before we start assigning values .

Reply:
Got it.